excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/AbstractDisplayGroupParser.java

```diff
@@ -141,9 +139,28 @@ public AbstractDisplayGroupParser(ParseContext parseContext, ShowConditionParser
 
         group.addDisplayGroupCaseFields(groupCaseFields);
         this.groupShowConditionColumn.ifPresent(column -> parseGroupShowCondition(column, group, groupDefinition.getValue()));
+        this.userRoleColumn.ifPresent(column -> parseUserRole(caseType, group, groupDefinition.getValue()));
         return ParseResult.Entry.createNew(group);
     }
 
+    private void parseUserRole(CaseTypeEntity caseType,
+                               final DisplayGroupEntity group,
+                               final  List<DefinitionDataItem> groupDefinition) {
+        if (groupDefinition.stream().filter(ddi -> ddi.getString(USER_ROLE) != null).count() > 1) {
+            throw new MapperException(
+                String.format("Please provide one user role row per tab in worksheet %s on column USER_ROLE for the tab %s",
+                    groupDefinition.get(0).getSheetName(), group.getReference()));
+        }
+        Optional<DefinitionDataItem> optionalDDI = groupDefinition.stream().filter(ddi -> ddi.getString(USER_ROLE) != null).findFirst();
+        if (optionalDDI.isPresent()) {
+            final String userRole = optionalDDI.get().getString(USER_ROLE);
+            UserRoleEntity roleEntity = parseContext.getRole(caseType.getReference(), userRole).orElseThrow(() ->
```

Comment:
can we extract this block of code into a private method with a self explanatory name?

Reply:
done

Reply:
done